integration/presto/src/main/java/org/apache/carbondata/presto/CarbondataConnectorFactory.java

```diff
@@ -17,69 +17,177 @@
+          injector.getInstance(ConnectorNodePartitioningProvider.class);
+      HiveSessionProperties hiveSessionProperties =
+          injector.getInstance(HiveSessionProperties.class);
+      HiveTableProperties hiveTableProperties = injector.getInstance(HiveTableProperties.class);
+      ConnectorAccessControl accessControl =
+          new PartitionsAwareAccessControl(injector.getInstance(ConnectorAccessControl.class));
+      Set<Procedure> procedures = injector.getInstance(Key.get(new TypeLiteral<Set<Procedure>>() {
+      }));
 
-      return new CarbondataConnector(lifeCycleManager,
-          new ClassLoaderSafeConnectorMetadata(metadata, classLoader),
-          new ClassLoaderSafeConnectorSplitManager(splitManager, classLoader), classLoader,
-          new ClassLoaderSafeConnectorPageSourceProvider(connectorPageSource, classLoader));
+      return new HiveConnector(lifeCycleManager, metadataFactory, transactionManager,
+          new ClassLoaderSafeConnectorSplitManager(splitManager, classLoader),
+          new ClassLoaderSafeConnectorPageSourceProvider(connectorPageSource, classLoader),
+          new ClassLoaderSafeConnectorPageSinkProvider(pageSinkProvider, classLoader),
+          new ClassLoaderSafeNodePartitioningProvider(connectorDistributionProvider, classLoader),
+          ImmutableSet.of(), procedures, hiveSessionProperties.getSessionProperties(),
+          HiveSchemaProperties.SCHEMA_PROPERTIES, hiveTableProperties.getTableProperties(),
+          accessControl, classLoader);
     } catch (Exception e) {
-      throw Throwables.propagate(e);
+      throwIfUnchecked(e);
+      throw new RuntimeException(e);
     }
   }
+
+  /**
+   * Set the Carbon format enum to HiveStorageFormat, its a hack but for time being it is best
+   * choice to avoid lot of code change.
+   *
+   * @throws Exception
+   */
+  private void setCarbonEnum() throws Exception {
+    for (HiveStorageFormat format : HiveStorageFormat.values()) {
+      if (format.name().equals("CARBON")) {
+        return;
+      }
+    }
+    Constructor<?>[] declaredConstructors = HiveStorageFormat.class.getDeclaredConstructors();
+    declaredConstructors[0].setAccessible(true);
+    Field constructorAccessorField = Constructor.class.getDeclaredField("constructorAccessor");
+    constructorAccessorField.setAccessible(true);
+    ConstructorAccessor ca =
+        (ConstructorAccessor) constructorAccessorField.get(declaredConstructors[0]);
+    if (ca == null) {
+      Method acquireConstructorAccessorMethod =
+          Constructor.class.getDeclaredMethod("acquireConstructorAccessor");
+      acquireConstructorAccessorMethod.setAccessible(true);
+      ca = (ConstructorAccessor) acquireConstructorAccessorMethod.invoke(declaredConstructors[0]);
+    }
+    Object instance = ca.newInstance(new Object[] { "CARBON", HiveStorageFormat.values().length, "",
+        "org.apache.carbondata.hadoop.api.CarbonFileInputFormat",
```

Comment:
should this be TableInputFormat ?

because in CarbondataPageSourceProvider.java we have used TableInputFormat

Reply:
please add assertion for serdes and outputformat class also